oak-segment-remote/src/main/java/org/apache/jackrabbit/oak/segment/remote/persistentcache/PersistentDiskCache.java

```diff
@@ -63,15 +63,10 @@
 
     final AtomicLong evictionCount = new AtomicLong();
 
-    private static final Comparator<Path> sortedByAccessTime = (path1, path2) -> {
-        try {
-            FileTime lastAccessFile1 = Files.readAttributes(path1, BasicFileAttributes.class).lastAccessTime();
-            FileTime lastAccessFile2 = Files.readAttributes(path2, BasicFileAttributes.class).lastAccessTime();
-            return lastAccessFile1.compareTo(lastAccessFile2);
-        } catch (IOException e) {
-            logger.error("A problem occurred while cleaning up the cache: ", e);
-        }
-        return 0;
+    private static final Comparator<SegmentCacheEntry> sortedByAccessTime = (segmentCacheEntry1, segmentCacheEntry2) -> {
```

Comment:
One more thing here: since you now have `SegmentCacheEntry` you could make it implement `Comparable` and move this logic in there. Then you need to call `sorted` on the stream below without arguments.

Reply:
Good suggestion, thanks!